src/Parsers/ParserSetQuery.cpp

```diff
@@ -109,7 +114,15 @@ bool ParserSetQuery::parseNameValuePair(SettingChange & change, IParser::Pos & p
         value = std::make_shared<ASTLiteral>(Field(static_cast<UInt64>(1)));
     else if (ParserKeyword("FALSE").ignore(pos, expected))
         value = std::make_shared<ASTLiteral>(Field(static_cast<UInt64>(0)));
-    else if (!value_p.parse(pos, value, expected))
+    /// for SETTINGS disk=disk(type='s3', path='', ...)
```

Comment:
That can be dangerous. I mean what if somebody will write `SETTINGS disk=disk(type='local', path='<some_system_path>')`

Reply:
Maybe it should accept only disk name, and `path` can be extracted from `ATTACH TABLE FROM <path>` (now it looks at `user_files`) ?

Reply:
> That can be dangerous. I mean what if somebody will write SETTINGS disk=disk(type='local', path='<some_system_path>')

and this is allowed, there are even tests for that, BUT it is allowed only if there is an explicit setting in server config that tells which base directory can be used for such out-of-config disks and in code all non-remote disks must lay in this directory.

Reply:
> Maybe it should accept only disk name, and path can be extracted from ATTACH TABLE FROM <path> (now it looks at user_files) ?

hm, this does not seem convenient for me